src/USER-BOCS/compute_pressure_bocs.cpp

```diff
@@ -122,6 +127,10 @@ ComputePressureBocs::~ComputePressureBocs()
   delete [] vector;
   delete [] vptr;
   if (phi_coeff) free(phi_coeff);
+
+  // Any splines data that was sent in from fix_bocs must be
+  // freed here, after it has been used.
+  memory->destroy(splines);
```

Comment:
This is very unusual and not a good idea. We need to correct this eventually. If the fix allocates the storage, it should free it. We have an established mechanism to share access with the "extract" function. The compute can use "extract" to retrieve pointers from the fix.

Reply:
Since the fix creates the compute, it can delete the splines data after the compute is deleted in the fix destructor.

Reply:
@akohlmey , I'd prefer to figure this out and get it right now since this is the core of the memory leak fix. 

I will try moving the delete to the fix destructor. At some point I had convinced myself that the data object might need  to outlive the fix object. But @mdelyser has a multi-threaded environment for testing. So, after I make the change to move the delete to the fix destructor, I will ask him to re-test.

If that simple change does not work safely, I will investigate the extract mechanism.

Reply:
Code has been moved to fix destructor and tested by me locally. I've committed and will request author @mdelyser test in his multithreaded environment.

Reply:
Original author has tested in his clustered environment and verified new destructor-based destroy works.